File: core/proj.android_studio/ee-core/src/main/java/com/ee/core/Logger.java
package com.ee.core;

import android.util.Log;

/**
 * Created by enrevol on 4/13/16.
 */
public class Logger {
    private String tag;

    public Logger(String tag) {
        super();
        this.tag = tag;
    }

    public void error(String msg) {
        Log.e(tag, msg);
    }

    public void error(String msg, Throwable th) {
        Log.e(tag, msg, th);
    }

    public void debug(String msg) {
        Log.d(tag, msg);
    }

    public void debug(String format, Object... objects) {
        Log.d(tag, String.format(format, objects));
    }

    public void info(String msg) {
        Log.i(tag, msg);
    }

    public void info(String format, Object... objects) {
        Log.i(tag, String.format(format, objects));
    }
}

File: core/src/Logger.cpp
//
//  PluginUtils.cpp
//  ee-core
//
//  Created by Zinge on 7/6/16.
//
//

#include "Logger.hpp"
#include "LogLevel.hpp"

namespace ee {
namespace core {
Logger::Logger(const std::string& tag)
    : tag_{tag}
    , enabled_{true} {}

void Logger::setEnabled(bool enabled) noexcept { enabled_ = enabled; }

bool Logger::isEnabled() const noexcept { return enabled_; }

void Logger::log(const LogLevel& level, const std::string& message) const {
    if (isEnabled()) {
        log(level, tag_, message);
    }
}

void Logger::debug(const std::string& message) const {
    log(LogLevel::Debug, message);
}

void Logger::info(const std::string& message) const {
    log(LogLevel::Info, message);
}

void Logger::warn(const std::string& message) const {
    log(LogLevel::Warn, message);
}

void Logger::error(const std::string& message) const {
    log(LogLevel::Error, message);
}
} // namespace core
} // namespace ee